configextractor.py

```diff
@@ -119,6 +119,8 @@ def section_builder(self, parser, field_dict, result, parsertype="MWCP"):
         if parsertype == "MWCP":
             for name, obj in self.file_parsers.items():
                 if parser in obj.parser_list:
+                    # TODO: we're constantly overwriting these values if there are hits
+                    # Is parser only in a single obj's parser_list?
```

Comment:
yes it should only ever be in a single parser_list

Reply:
because it doesn't make any sense to have a parser that ran in multiple entries in the yara_parser.yaml

Reply:
Okay great. Next question... after running debug, if looks like `obj.parser_list` is always a list with a single item which is equal to `name`, and `parser` is always equal to one of the keys in `self.file_parsers`, so why are we even iterating at all? Why are we checking the `obj.parser_list` attribute when the value contained in it is the same as `name`?  Why not just do `obj = self.file_parsers[parser]` ?

Reply:
not quite, parser_list is usually just a single item but it can be more. This is more for future reasons if you need to run multiple parsers on one yara rule hit. It's not really being used currently. parser is actually not always equal to one of the keys in self.file_parsers. There's two different names here. parser will always be one of the fields under the parser - MWCP block in the yaml. The keys in self.file_parsers are the names in the start of each block in yara_parser.yaml see here 
https://github.com/CybercentreCanada/assemblyline-service-configextractor/blob/master/yara_parser.yaml
Now they look the same here but in the proprietary version there are differences and this handles that. That's why it's like this.

Reply:
okay, then if that is the case, then I suggest that after we find a hit for `if parser in obj.parser_list` and we assign all of the variables, we break out of the `for` loop with `break` since there is no point of continuing to iterate through `self.file_parsers.items()` looking for hits given that there should only be one hit ever.

Reply:
done here
https://github.com/CybercentreCanada/assemblyline-service-configextractor/commit/4fedc0be87b62d6395a865bbf606daf5f839076a